eng/common/TestResources/New-TestResources.ps1

```diff
@@ -215,11 +215,18 @@ try {
     $PSBoundParameters['BaseName'] = $BaseName
 
     # Try detecting repos that support OutFile and defaulting to it
-    if (!$CI -and !$PSBoundParameters.ContainsKey('OutFile') -and $IsWindows) {
+    if (!$CI -and !$PSBoundParameters.ContainsKey('OutFile')) {
         # TODO: find a better way to detect the language
-        if (Test-Path "$repositoryRoot/eng/service.proj") {
+        if ($IsWindows -and (Test-Path "$repositoryRoot/eng/service.proj")) {
             $OutFile = $true
-            Log "Detected .NET repository. Defaulting OutFile to true. Test environment settings would be stored into the file so you don't need to set environment variables manually."
+            $isDotnet = $true
+            Log "Detected .NET repository. Defaulting OutFile to true. Test environment settings will be stored into a file so you don't need to set environment variables manually."
+        } elseif (Test-Path "$root/assets.json") {
+            $assets = Get-Content -Raw "$root/assets.json" | ConvertFrom-Json
+            if ($assets.Dotenv) {
```

Comment:
So you're imagining boolean value here:

```json
{
  "AssetsRepo": "Azure/azure-sdk-assets",
  "AssetsRepoPrefixPath": "net",
  "TagPrefix": "net/eventgrid/Azure.ResourceManager.EventGrid",
  "Tag": "net/eventgrid/Azure.ResourceManager.EventGrid_442af20ea8",
  "Dotenv": true
}
```
The test proxy should be able to handle this. I will double check that it doesn't throw if we add this.

If anything, we should remove the containing `IsWindows` and `eng/service.proj` path, because if the assets.json has `Dotenv` present then we should save the .env without caring about the other possible values.

Reply:
[Looks good locally but double checking integration tests](https://github.com/Azure/azure-sdk-tools/pull/10327)

Reply:
Well, the `$IsWindow` check and friends are there for back compat. On Windows, it has long supported a *custom* `.env`-like file, but it's JSON-encoded and then DPAPI-encryped, which is why it's only on Windows. And it's only for Azure/azure-sdk-for-net because no other language cared to do the work to call DPAPI which, in .NET, has a convenient API - at least on Windows.

So, I don't want to break that. In fact, I wanted to get this PR in because I miss what I had in .NET on Windows (for .NET, VS > VSCode).

That's also why I do that check first. I want to maintain backcompat so even if a library on Windows authors `"Dotenv": true`, I'd think @JoshLove-msft and folk would probably want to maintain backcompat.

Reply:
Ok fair point. I don't oppose this PR then. Wes is out this week so @benbp for FYI.